lib/codegen/analysis/align.cc

```diff
@@ -103,15 +103,15 @@ std::vector<align::cst_info> align::populate_is_constant_reshape(ir::reshape_ins
     cst_info ax ;
     if(x_shapes[d] == 1)
       ax = {1, op_cst[current].value};
-    else if(!is_skewed
-            && x_shapes[d] == op_shapes[current])
+    else if (!is_skewed && current < op_shapes.size() &&
```

Comment:
Shouldn't the bounds check be on `op_cst`?

Reply:
On this condition, it should be on `op_shapes`. 

On the condition above (i.e.,  x_shapes[d] == 1), it should be on `op_cst` but I'm not sure whether current will be out of bound when `x_shapes[d] == 1`. @ptillet definitely knows more about the details.